components/pluralrules/src/rules/ast.rs

```diff
@@ -0,0 +1,314 @@
+///     },
+///     Relation {
+///         expression: Expression {
+///             operand: Operand::V,
+///             modulus: None,
+///         },
+///         operator: Operator::NotIn,
+///         range_list: RangeList(Box::new([RangeListItem::Value(Value(2))])),
+///     },
+/// ]));
+///
+/// ```
+#[derive(Debug, Clone, PartialEq)]
+pub struct AndCondition(pub Box<[Relation]>);
+
+/// An incomplete AST representation of a plural rule. Comprises an Expression, an Operator, and a RangeList.
+///
+/// # Examples
+///
+/// All AST nodes can be built explicitly, as seen in the example. However, due to its complexity, it is preferred to build the AST using the parse_plural_rule function.
+///
+/// ```text
+/// "i = 3"
+/// ```
+///
+/// Can be represented by the AST:
+///
+/// ```
+/// use icu_pluralrules::rules::ast::*;
+///
+/// Relation {
+///     expression: Expression {
+///         operand: Operand::I,
+///         modulus: None,
+///     },
+///     operator: Operator::Is,
+///     range_list: RangeList(Box::new([RangeListItem::Value(Value(3))])),
+/// };
+///
+/// ```
+#[derive(Debug, Clone, PartialEq)]
+pub struct Relation {
+    pub expression: Expression,
+    pub operator: Operator,
+    pub range_list: RangeList,
+}
+
+/// An enum of Relation operators for plural rules.
+///
+/// Each Operator enumeration belongs to the corresponding symbolic operators:
+///
+/// | Enum Operator | Symbolic Operator |
+/// | - | - |
+/// | In | "in" |
+/// | NotIn | "not in" |
+/// | Within | "within" |
+/// | NotWithin | "not within" |
+/// | Is | "is" |
+/// | IsNot | "is not" |
+/// | Eq | "=" |
+/// | NotEq | "!=" |
+///
+#[derive(Debug, Clone, PartialEq)]
+pub enum Operator {
+    In,
+    NotIn,
+    Within,
+    NotWithin,
+    Is,
+    IsNot,
+    Eq,
+    NotEq,
+}
+
+/// An incomplete AST representation of a plural rule. Comprises an Operand and an optional Modulo.
+///
+/// # Examples
+///
+/// All AST nodes can be built explicitly, as seen in the example. However, due to its complexity, it is preferred to build the AST using the parse_plural_rule function.
+///
+/// ```text
+/// "i % 100"
+/// ```
+///
+/// Can be represented by the AST:
+///
+/// ```
+/// use icu_pluralrules::rules::ast::*;
+///
+/// Expression {
+///     operand: Operand::I,
+///     modulus: Some(Value(100)),
+/// };
+///
+/// ```
+#[derive(Debug, Clone, PartialEq)]
+pub struct Expression {
+    pub operand: Operand,
+    pub modulus: Option<Value>,
+}
+
+/// An incomplete AST representation of a plural rule. Comprises a char.
+///
+/// # Examples
+///
+/// All AST nodes can be built explicitly, as seen in the example. However, due to its complexity, it is preferred to build the AST using the parse_plural_rule function.
+///
+/// ```text
+/// "i"
+/// ```
+///
+/// Can be represented by the AST:
+///
+/// ```
+/// use icu_pluralrules::rules::ast::Operand;
+///
+/// Operand::I;
+/// ```
+#[derive(Debug, Clone, PartialEq)]
+pub enum Operand {
```

Comment:
If `Operand` here is merely the AST node that represents the occurrence of an operand identifier within a rule string (and the rule string is what the full AST is for), then maybe the comments below describing the meaning of N, I, V, W, ... are actually misleading.  Maybe instead, you might want comments like `/// AST node for plural operand N occurring within a plural rule string` ?

Reply:
I changed the docstring to indicate its an AST node. Hope that helps!

Reply:
Yep, this should be good now.